presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/SpillableHashAggregationBuilder.java

```diff
@@ -70,8 +77,13 @@ public SpillableHashAggregationBuilder(
         this.hashChannel = hashChannel;
         this.operatorContext = operatorContext;
         this.memorySizeBeforeSpill = memoryLimitBeforeSpill.toBytes();
+        this.memoryLimitForMergeWithMemory = memoryLimitForMergeWithMemory.toBytes();
         this.spillerFactory = spillerFactory;
-        this.systemMemoryContext = operatorContext.getSystemMemoryContext().newLocalMemoryContext();
+
+        AbstractAggregatedMemoryContext systemMemoryContext = operatorContext.getSystemMemoryContext();
+        this.aggregationMemoryContext = systemMemoryContext.newLocalMemoryContext();
+        this.spillMemoryContext = systemMemoryContext.newLocalMemoryContext();
+        this.mergeMemoryContext = systemMemoryContext.newLocalMemoryContext();
```

Comment:
don't make a field but generate it on the fly during construction of `MergingAggregationBuilder`